grpc.py

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python3
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+import json
+import re
+import subprocess
+
+
+def parse_description(description):
+    """
+    Parses the description of an RPC method.
+
+    Example input:
+        /** lncli: `getinfo`
+        GetInfo serves a request.
+        It returns info.
+        */
+
+    Example output:
+        ('getinfo', "GetInfo serves a request. It returns info.")
+    """
+
+    if description is None:
+        return None, None
+
+    # Match lncli: `lncli_name` or lncli:`lncli_name`.
+    command = None
+    match = re.search(r'(?:lncli:.?`)(.*)(?:`)', description)
+    if match:
+        command = match.group(1)
+        description = description[match.end():].strip()
+
+    # Replace all new lines so that Slate markdown renders correctly.
+    description = description.replace('\n', ' ').replace('\r', '')
+
+    return command, description
+
+
+def parse_grpc_message_params(message):
+    """
+    Parses the parameters of a gRPC message and returns them as a list.
+    """
+
+    params = []
+    for param in message['fields']:
+        _, description = parse_description(param['description'])
+
+        p = {
+            'name': param['name'],
+            'description': description
+        }
+
+        if param['label'] == 'repeated':
+            p['type'] = 'array ' + param['type']
+        else:
+            p['type'] = param['type']
+
+        # If the type includes an "lnrpc." prefix, then we'll remove it in order
+        # to include a link to the message being referenced.
+        lnrpc_prefix = "lnrpc."
+        if param['fullType'][:len(lnrpc_prefix)] == lnrpc_prefix:
```

Comment:
Not a blocker, but could instead use `.startswith(..)` here.

Reply:
Fixed.